### PR TITLE
update members data api to play 2.8 plus other bumps

### PR DESCRIPTION
**Pull request by johnduffell**
Identity release the libraries mainly for play 2.8 now, the 2.7 process is a bit of a bolt on on a special branch.
So I have decided the easiest solution is to bump members data api to play 2.8 as it needs doing anyway.

This will make it easier to get the snyk updates out in https://github.com/guardian/members-data-api/pull/571

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -22,37 +22,37 @@ def buildInfoSettings = Seq(
...
+  scalaVersion := "2.12.14",
   resolvers ++= Seq(
     "Guardian Github Releases" at "https://guardian.github.io/maven/repo-releases",
     "Guardian Github Snapshots" at "https://guardian.github.io/maven/repo-snapshots",
     Resolver.sonatypeRepo("releases")
   ),
-  sources in (Compile, doc) := Seq.empty,
-  publishArtifact in (Compile, packageDoc) := false,
-  parallelExecution in Global := false,
+  Compile / doc / sources := Seq.empty,
```
Comment (johnduffell) [inline build.sbt:31]: these are needed because the "in" syntax is deprecated in sbt 1.5

File: build.sbt [hunk 2]
```diff
@@ -74,7 +74,10 @@ def app(name: String) =
     .settings(buildDebSettings)
 
 val api = app("membership-attribute-service")
-  .settings(libraryDependencies ++= apiDependencies)
+  .settings(
+    libraryDependencies ++= apiDependencies,
+    dependencyOverrides ++= depOverrides,
+  )
```
Comment (johnduffell) [inline build.sbt:80]: aws client brings in 2.12 of jackson-databind, but plaay 2.8 brings in 2.11.  Both of those aare security patched, but 2.11 is not happy to use the 2.12 jars at runtime.
```
com.typesafe.play:play_2.12:2.8.8                  (depends on 2.11.4)
software.amazon.awssdk:sdk-core:2.16.69            (depends on 2.12.3)
```
however now it just bombs out at runtime if there is a  binary conflict
`com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.11.4 requires Jackson Databind version >= 2.11.0 and < 2.12.0`

**Review by tomrf1 — APPROVED** [review]
great work!

**Comment by johnduffell** [comment]
this wouldn't start in CODE because of this error:

```
Jun 15 11:55:37 ip-10-248-135-150 membership-attribute-service[5052]: Oops, cannot start the server.
Jun 15 11:55:37 ip-10-248-135-150 membership-attribute-service[5052]: Configuration error: Configuration error[
Jun 15 11:55:37 ip-10-248-135-150 membership-attribute-service[5052]: The application secret has not been set, and we are in prod mode. Your application is not secure.
Jun 15 11:55:37 ip-10-248-135-150 membership-attribute-service[5052]: To set the application secret, please read http://playframework.com/documentation/latest/ApplicationSecret
Jun 15 11:55:37 ip-10-248-135-150 membership-attribute-service[5052]:           ]
```

It turned out we were still using the (deprecated since 2.6) 

play.crypto.secret instead of play.http.secret.key (see https://www.playframework.com/documentation/2.8.x/Migration26#Other-Configuration-changes )

I have changed it in the CODE and PROD config and CODE is starting fine now.

**Comment by johnduffell** [comment]
everything looks good overall, although weirdly we stopped ELB 5xx at the deploy time
image

**Comment by johnduffell** [comment]
oddly they only started recently, maybe one of the instances had an issue
image
https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#metricsV2:graph=~(metrics~(~(~'AWS*2fELB~'HTTPCode_ELB_5XX~'LoadBalancerName~'Memb-Attr-LoadBala-1MU0G8YTTF0C1))~view~'timeSeries~stacked~false~region~'eu-west-1~stat~'Sum~period~60~yAxis~(left~(min~0))~start~'-P1D~end~'P0D);query=~'*7bAWS*2fELB*2cLoadBalancerName*7d*20elb*205xx

**Pull request merged**